src/presign/participant.rs

```diff
@@ -239,7 +199,12 @@ impl PresignParticipant {
     /// These round two messages are returned in response to the sender, without having to
     /// rely on any other round one messages from other participants aside from the sender.
     #[cfg_attr(feature = "flame_it", flame)]
-    fn do_round_two(&mut self, message: &Message, main_storage: &Storage) -> Result<Vec<Message>> {
+    fn handle_round_one_msg<R: RngCore + CryptoRng>(
```

Comment:
This function has been renamed to `handle_round_one_msg` but the documentation still refers to round two? Does this need to be updated/changed?

Reply:
This is a good point and sort of a weird case. Generally for these protocols, for each round there's a handle_round_n_msg() function which validates messages, stores anything relevant, and possibly calls gen_round_n+1_msgs() which returns a vec of messages to send out.

In this instance, the appropriate round 2 message can be sent to a Participant after receiving that Participant's round 1 message. Maybe the way to handle this is have handle_round_one_msg call a new function called gen_round_two_msg(). As it is right now though, all handle_round_one_msg() would do is immediately call gen_round_two_msg() as there aren't really any checks that are easily factored out. In handling Issue #11 next though, I'll add checks about whether or not it's possible to process this message yet, and breaking apart the functions like this will make more sense.

Maybe we save this for that PR?

Reply:
I think the current structure is reasonable, where `handle_round_n` produces output messages for round `n+1`. I'm a little loath to start an overhaul of code structure like that, because validation (for round n) is rarely completely independent from processing (into round n+1) -- although we'll see how things look in the next PR. I recommend updating the documentation to more clearly explain the context here. Since the first line of docs is what shows up in IDEs, you could say something like
> Processes a single request from round one to create public keyshares for that participant, to be sent in round two.

Then you can keep the rest of the docs the same, they add useful context.